### PR TITLE
feat: Implement 'update' method for MariaDB clusters

### PR DESCRIPTION
**Pull request by adeatcu-ionos**
## Checklist

- [ ] PR name added as appropriate (e.g. `feat:`/`fix:`/`doc:`/`test:`/`refactor:`)
- [ ] Tests added or updated
- [ ] Documentation updated
- [ ] Changelog updated and version incremented (label: upcoming release)
- [ ] Github Issue linked if any
- [ ] Jira task updated

### CODE REVIEW

**Comment by adeatcu-ionos** [comment]
First commit: SDK was generated locally, the `version` feature it's not released yet in the API.

File: services/dbaas/mariadb/cluster.go [hunk 1]
```diff
@@ -158,6 +167,65 @@ func GetMariaDBClusterDataCreate(d *schema.ResourceData) (*mariadb.CreateCluster
...
+		newValueString := newValue.(string)
+		oldVersion, err := goVersion.NewVersion(oldValueString)
+		if err != nil {
+			return nil, err
+		}
+		newVersion, err := goVersion.NewVersion(newValueString)
+		if err != nil {
+			return nil, err
+		}
+		if newVersion.LessThan(oldVersion) {
```
Comment (cristiGuranIonos) [inline services/dbaas/mariadb/cluster.go:187]: add as validation function if possible

Comment (cristiGuranIonos) [inline services/dbaas/mariadb/cluster.go:187]: add description for version that it cannot be downgraded

File: ionoscloud/resource_dbaas_mariadb_cluster.go [hunk 2]
```diff
@@ -254,5 +254,25 @@ func mariaDBClusterRead(ctx context.Context, d *schema.ResourceData, meta interf
...
+
+	clusterID := d.Id()
+	cluster, err := mariadb.GetMariaDBClusterDataUpdate(d)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	response, _, err := client.UpdateCluster(ctx, *cluster, clusterID, d.Get("location").(string))
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("an error occurred while updating DBaaS MariaDB cluster with ID: %v, error: %w", clusterID, err))
```
Comment (cristiGuranIonos) [inline ionoscloud/resource_dbaas_mariadb_cluster.go:267]: add location to err

File: ionoscloud/resource_dbaas_mariadb_cluster.go [hunk 3]
```diff
@@ -254,5 +254,25 @@ func mariaDBClusterRead(ctx context.Context, d *schema.ResourceData, meta interf
...
+	}
+
+	err = utils.WaitForResourceToBeReady(ctx, d, client.IsClusterReady)
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("error occurred while checking the status for MariaDB cluster with ID: %v, error: %w", clusterID, err))
```
Comment (cristiGuranIonos) [inline ionoscloud/resource_dbaas_mariadb_cluster.go:272]: same

**Review by rmocanu-ionos — DISMISSED** [review]

**Review by rmocanu-ionos — APPROVED** [review]

**Pull request merged**